1-js/06-advanced-functions/02-rest-parameters-spread/article.md

```diff
@@ -118,49 +118,49 @@ function f() {
 f(1); // 1
 ```
 
-As we remember, arrow functions don't have their own `this`. Now we know they don't have the special `arguments` object either.
+همانطور که به خاطر داریم، توابع تک خطی از خود `this` ندارند. اکنون می‌دانیم که آنها شی `arguments` خاصی نیز ندارند.
 ````
 
 
-## Spread syntax [#spread-syntax]
+## سینتکس اسپرد [#spread-syntax]
 
-We've just seen how to get an array from the list of parameters.
+ما فقط نحوه گرفتن یک آرایه از لیست پارامترها را دیدیم.
 
-But sometimes we need to do exactly the reverse.
+اما گاهی اوقات ما باید دقیقاً برعکس عمل کنیم.
 
-For instance, there's a built-in function [Math.max](mdn:js/Math/max) that returns the greatest number from a list:
+برای مثال، یک تابع داخلی [Math.max](mdn: js/Math/max) وجود دارد که بیشترین عدد را از یک لیست بر می‌گرداند:
 
 ```js run
 alert( Math.max(3, 5, 1) ); // 5
 ```
 
-Now let's say we have an array `[3, 5, 1]`. How do we call `Math.max` with it?
+حالا فرض کنید یک آرایه داریم `[3, 5, 1]`. چگونه با آن تابع `Math.max` را فراخوانی کنیم؟
 
-Passing it "as is" won't work, because `Math.max` expects a list of numeric arguments, not a single array:
+وارد کردن آن "همانطور که هست" کار نخواهد کرد، زیرا `Math.max` لیستی از آرگومان های عددی را انتظار دارد، نه یک آرایه واحد:
 
 ```js run
 let arr = [3, 5, 1];
 
 *!*
 alert( Math.max(arr) ); // NaN
-*/!*
+*/!*  
 ```
 
-And surely we can't manually list items in the code `Math.max(arr[0], arr[1], arr[2])`, because we may be unsure how many there are. As our script executes, there could be a lot, or there could be none. And that would get ugly.
+و مطمئناً ما نمی‌توانیم مواردی را در تابع `Math.max (arr [0], arr [1], arr [2])` به صورت دستی لیست کنیم، زیرا ممکن است مطمئن نباشیم تعداد آنها چقدر است. همانطور که اسکریپت ما اجرا می‌شود، ممکن است تعداد زیادی (آرگومان) وجود داشته باشد، یا ممکن است وجود نداشته باشد. و این باعث زشتی کد ما می‌شود.
```

Comment:
You shouldn't change the codes since it'll cause conflicts:
```suggestion
و مطمئناً ما نمی‌توانیم مواردی را در تابع `Math.max(arr[0], arr[1], arr[2])` به صورت دستی لیست کنیم، زیرا ممکن است مطمئن نباشیم تعداد آنها چقدر است. همانطور که اسکریپت ما اجرا می‌شود، ممکن است تعداد زیادی (آرگومان) وجود داشته باشد، یا ممکن است وجود نداشته باشد. و این باعث زشتی کد ما می‌شود.
```